### DISCUSSION TITLE
Page with dataview query is not updated when the dataview query is refreshed

### DISCUSSION THREAD

**Discussion by h7b** [ObsidianPublisher/obsidian-github-publisher#129] [Q&A — Plugin] [answered]
Hi,

When using the GitHub Publisher plugin, I found that a page contained a dataview query is not marked as "updated" when the dataview query is refreshed. This leads to the content of this page is not refreshed by the plugin

## What I did

In my Obsidian vault, I have a file `dataview.md` with the body content as below, and the key "share: true" in frontmatter. When I add new note into folder "daily-note", the dataview will be refreshed.

````
```dataview
TABLE ticker AS ticker, 
orderType AS order, 
price AS price, 
quantity AS quantity

FROM "daily-notes"
WHERE ticker != NULL
SORT file.ctime DESC
```
````

## What I expect

When I use the command "GitHub Publisher: Refresh published and upload new notes", the `dataview.md` will be marked as `updated` and will be uploaded/refreshed in the github repo

But it does not, the file `dataview.md` with refreshed content from dataview query, is not uploaded/refreshed in the repo

## Current workaround

I need to edit the H1 heading of the `dataview.md` file, then revert the change. After this step, the command "GitHub Publisher: Refresh published and upload new notes" will marked the file `dataview.md` as `updated` and send the file to repo, as expected

I don't have the expertise to understand the code yet. But I found that the plugin [digital-garden](https://github.com/oleeskild/obsidian-digital-garden) can refresh the content of dataview query when publishing pages, without the needs of manually edit the H1 heading as I did with GitHub Publisher plugin. 

I hope you have the interest to research what was happening with the dataview conversion part.

**Comment by Mara-Li** [comment]
Personally, when I edit my dataview I just use the share one file command, so no really need to change the h1 tbh.

Also, my way to check if a file is updated in Obsidian is to compare the edited time with the last push time.

But dataview don't edit the content so obviously the edited time doesn't change.

Digital garden and my implementation is too much different to just take the same idea.

**Reply by h7b** [reply]
aha, thanks for explaining the mechanism of checking file. Now i understand that's the expected behavior of the plugin's dataview conversion part.

If you ever thought about making the plugin always refresh/re-upload the content of a dataview query, it would be a lovely hack. Though I don't understand how much is the computing cost of this hack.

In my experiment, the dataview conversion makes the mkdocs-material buitl-in tags plugin become redundant and not flexible enough :D

**Reply by Mara-Li** [reply]
Maybe a plug-in or something can be trick an edit on the dataview note when updated. Like a frontmatter "updated: <date>". I think it must hijack the refreshing of a dataview table/list/etc. 

To be honest I don't know how to do that ^^'

**Comment by Mara-Li** [comment, marked as answer]
I updated the docs to add informations about dataview & refreshed commands.

**Reply by h7b** [reply]
https://obsidian-publisher.netlify.app/github%20publisher/commands/?h=dataview#refresh

Yay. I saw the update in docs. At the moment, it means that when a note has a dataview query, we must use the "Upload" type command to resend the note, whenever we think that the output of the dataview query might have been updated

Hope we can have a more elegant approach in the future
